cli/azd/pkg/azsdk/storage/storage_blob_client.go

```diff
@@ -0,0 +1,240 @@
+
+		for _, blob := range page.Segment.BlobItems {
+			blobs = append(blobs, &Blob{
+				Name:         filepath.Base(*blob.Name),
+				Path:         *blob.Name,
+				CreationTime: *blob.Properties.CreationTime,
+				LastModified: *blob.Properties.LastModified,
+			})
+		}
+	}
+
+	return blobs, nil
+}
+
+// Download downloads a blob from the configured storage account container.
+func (bc *blobClient) Download(ctx context.Context, blobPath string) (io.ReadCloser, error) {
+	client, err := bc.createClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return nil, err
+	}
+
+	resp, err := client.DownloadStream(ctx, bc.config.ContainerName, blobPath, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download blob '%s', %w", blobPath, err)
+	}
+
+	return resp.Body, nil
+}
+
+// Upload uploads a blob to the configured storage account container.
+func (bc *blobClient) Upload(ctx context.Context, blobPath string, reader io.Reader) error {
+	client, err := bc.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return err
+	}
+
+	_, err = client.UploadStream(ctx, bc.config.ContainerName, blobPath, reader, nil)
+	if err != nil {
+		return fmt.Errorf("failed to upload blob '%s', %w", blobPath, err)
+	}
+
+	return nil
+}
+
+// Delete deletes a blob from the configured storage account container.
+func (bc *blobClient) Delete(ctx context.Context, blobPath string) error {
+	client, err := bc.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return err
+	}
+
+	_, err = client.DeleteBlob(ctx, bc.config.ContainerName, blobPath, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete blob '%s', %w", blobPath, err)
+	}
+
+	return nil
+}
+
+// Check if the specified container exists
+// If it doesn't already exist then create it
+func (bc *blobClient) ensureContainerExists(ctx context.Context) error {
+	client, err := bc.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	exists := false
+
+	pager := client.NewListContainersPager(nil)
+	for pager.More() {
+		if exists {
+			break
+		}
+
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed getting next page of containers: %w", err)
+		}
+
+		for _, container := range page.ContainerItems {
+			if *container.Name == bc.config.ContainerName {
+				exists = true
+				break
+			}
+		}
+	}
+
+	if !exists {
+		_, err = client.CreateContainer(ctx, bc.config.ContainerName, nil)
+		if err != nil {
+			return fmt.Errorf("failed to create container '%s', %w", bc.config.ContainerName, err)
+		}
+	}
+
+	return nil
+}
+
+// createClient creates a new blob client and caches it for future use
+func (bc *blobClient) createClient(ctx context.Context) (*azblob.Client, error) {
+	if bc.client != nil {
+		return bc.client, nil
+	}
+
+	credential, err := bc.authManager.CredentialForCurrentUser(ctx, nil)
```

Comment:
@vhvb1989 is there some limitation with `CredentialForCurrentUser` that it would only restrict to tenant owners?  I am currently under the impression that it would use the permissions of the logged in user when reading/writing to blob storage vs requiring a different authentication strategy of manual connection of SAS tokens.

Reply:
The nice thing about using managed identity here is that we don't have to worry about storing any secrets in our configuration.  I think in practice that folks are not going to use Anonymous access here (especially since you'd have to allow anonymous writes, which is super scary).  Perhaps supporting SAS as well would be reasonable, but then we'd have this problem of securing the SAS token itself, which we don't have a great story for.

The downside is that Azure RBAC is a little tricky, and so you might find that you need to grant RBAC permissions to the principal that is doing the deployment.  In an ideal world, it might be nice for these to be separatable, but I feel like starting with just a "you must use managed identity and Azure RBAC" is fine.  If it ends up being such an issue that we have to figure out how to go add support for SAS, I think there's a reasonable path forward for us to do this.